devito/types/grid.py

```diff
@@ -702,11 +708,12 @@ def __init__(self, **kwargs):
         self._global_bounds = kwargs.get('bounds', None)
 
     def __subdomain_finalize__(self, grid, **kwargs):
+        self._grid = grid
         self._dtype = grid.dtype
 
         # Create the SubDomainSet SubDimensions
         self._dimensions = tuple(
-            MultiSubDimension('%si_%s' % (d.name, self.implicit_dimension.name), d)
+            MultiSubDimension('%si_%s' % (d.name, self.implicit_dimension.name), d, self)
```

Comment:
That's some odd circular dependent thing here no?

Reply:
I agree was just wondering but don't have a way around neither looking at it. Maybe just add a TODO/Comment for later

Reply:
will do now